### PR TITLE
Topic more help fixes

### PR DESCRIPTION
**Pull request by adcxyz**
Here is another batch of mainly little fixes to help file render issues. 
Most of them are trivial, possibly some discussion is needed for: 
* the arg name changes in TextView (I found no uses of them with arg names)
* removing much obsolete info in GUI (should be its own issue)
* temp-adding stubs for classes and methods planned for the Image class, see #2605 

code::
// In any case, 
SCDoc.renderAll;
// now results in:
SCDoc: Rendering all documents
SCDoc: Indexing help-files...
SCDoc: Indexed 1338 documents in 0.71 seconds
WARNING: SCDoc: In /Users/adc/scgit/supercollider/HelpSource/Classes/MIDIClient.schelp
  Method *getClientID not found.
SCDoc: Done!
-> SCDoc
::
MIDIClient:getClientID exists in linux only - while one could move the doc to a linux-only folder, I think it is still useful to look up the method doc  on osx or windows.

### CODE REVIEW

File: HelpSource/Classes/GUI.schelp [hunk 1]
```diff
@@ -7,114 +7,17 @@ related:: Overviews/GUI-Classes, Guides/GUI-Introduction
 description::
 SuperCollider currently supports three operating system platforms: macOS, UNIX (Linux and FreeBSD) and Windows (with some limitations).
 
-warning::The redirect system has been deprecated, please use the view classes directly.::
+Warning::The redirect system has been deprecated, please use the view classes directly. If you find old code that uses prefixes for the old GUI Kits (e.g. SCWindow for CocoaGUI, JSCWindow for SwingOSC), try dropping the prefixes to Window, Button etc. The Qt prefix Q, as e.g. in QWindow, is converted to Window automatically; still it is better style to simple write Window.::
```
Comment (vivid-synth) [inline HelpSource/Classes/GUI.schelp]: s/simple write Window/simpl**y** write window/

File: HelpSource/Classes/GUI.schelp [hunk 2]
```diff
@@ -133,11 +36,13 @@ margin: 2@2,
 buttonHeight: 16
 ::
 
-method:: cocoa
-Makes Cocoa (macOS GUI) the current scheme and returns it. Subsequent GUI object calls to GUI are delegated to cocoa. Returns the current (cocoa) scheme.
+method:: add
+
+method:: qt
+Makes QtGUI the current scheme and returns it. Subsequent GUI object calls to GUI are delegated to swing. Returns the current (swing) scheme.
```
Comment (vivid-synth) [inline HelpSource/Classes/GUI.schelp]: "swing" is wrong here -- should be "Qt"

File: HelpSource/Classes/GUI.schelp [hunk 3]
```diff
@@ -133,11 +36,13 @@ margin: 2@2,
...
 
 method:: swing
-Makes Swing (Java GUI) the current scheme and returns it. Subsequent GUI object calls to GUI are delegated to swing. Returns the current (swing) scheme.
+If SwingOSC is installed, makes SwingGUI the current scheme and returns it. Subsequent GUI object calls to GUI are delegated to swing. Returns the current (swing) scheme.
```
Comment (vivid-synth) [inline HelpSource/Classes/GUI.schelp]: I think the parenthetical "(Java GUI)" is helpful

File: HelpSource/Classes/ScrollView.schelp [hunk 4]
```diff
@@ -36,77 +36,69 @@ SUBSECTION:: Geometry
...
-    Defaults to strong::true::.
+Defaults to strong::true::.
 
 METHOD:: hasVerticalScroller
 
-    Sets or gets whether the view has the vertical scrollbar. If this is code::true::, the scrollbar may still be hidden if link::#-autohidesScrollers:: allows so; however, if this it code::false:: the scrollbar will never be shown.
-
-    Defaults to strong::true::.
-
-METHOD:: autoScrolls
```
Comment (vivid-synth) [inline HelpSource/Classes/ScrollView.schelp]: autoScrolls doesn't exist?

File: HelpSource/Classes/ScrollView.schelp [hunk 5]
```diff
@@ -123,22 +115,22 @@ var layout;
...
-    i = i + 1;
-    view;
+	var view = View().background_(Color.rand).layout_(
+		HLayout(
+			TextField().string_( ("This is entry number " + i.asString) ),
+			Button().states_([["Delete"]]).action_({view.remove; i = i - 1;})
+		)
+	);
+	i = i + 1;
+	view;
```
Comment (vivid-synth) [inline HelpSource/Classes/ScrollView.schelp]: Do we definitely prefer tabs?

File: HelpSource/Classes/ScrollView.schelp (hunk 4 above)
Comment (vivid-synth) [inline HelpSource/Classes/ScrollView.schelp]: Nevermind, I see the comment below!

File: SCClassLibrary/Common/GUI/Base/QTextView.sc [hunk 6]
```diff
@@ -9,35 +9,35 @@ TextView : QAbstractScroll {
...
+	string_ { arg string;
+		this.setProperty( \plainText, string );
 	}
 
-	font_ { arg aFont;
-		font = aFont;
-		this.setProperty( \textFont, aFont );
+	font_ { arg argFont;
+		font = argFont;
+		this.setProperty( \textFont, font );
```
Comment (vivid-synth) [inline SCClassLibrary/Common/GUI/Base/QTextView.sc]: should be "argFont"

File: SCClassLibrary/JITLib/GUI/EnvirGui.sc [hunk 7]
```diff
@@ -80,8 +80,8 @@ EnvirGui : JITGui {
 		options.do { |key| extraFuncs[key].value; };
 	}
 
-	makeNameView { |nameWid, height|
-		nameView = StaticText(zone, Rect(0,0, nameWid, height))
+	makeNameView { |width, height|
+		nameView = StaticText(zone, Rect(0,0, width, height))
 			.font_(font).align_(0);
```
Comment (vivid-synth) [inline SCClassLibrary/JITLib/GUI/EnvirGui.sc]: Where does "font" come from here?

File: HelpSource/Classes/GUI.schelp (hunk 1 above)
Comment (adcxyz) [inline HelpSource/Classes/GUI.schelp]: OK, done

File: HelpSource/Classes/GUI.schelp (hunk 3 above)
Comment (adcxyz) [inline HelpSource/Classes/GUI.schelp]: OK, done

File: SCClassLibrary/Common/GUI/Base/QTextView.sc (hunk 6 above)
Comment (adcxyz) [inline SCClassLibrary/Common/GUI/Base/QTextView.sc]: font is an instance var of View, and after doing font = argFont, 
the two are the same object. 
I find setting to the current state of the instvar clearer than 
referring to the temporary method argument.

File: SCClassLibrary/JITLib/GUI/EnvirGui.sc (hunk 7 above)
Comment (adcxyz) [inline SCClassLibrary/JITLib/GUI/EnvirGui.sc]: it is an instvar of View

**Comment by telephon** [comment]
looks good to me

**Comment by brianlheim** [comment]
Would it make more sense for `MIDIClient.getClientID` to throw a `shouldNotImplement` error on non-Linux systems?

**Comment by adcxyz** [comment]
AFAICT it should only ever be called on linux, so we could implement it as a no-op with a little post:
```
getClientID { 
	if (thisProcess.platform != \linux) { 
		"% should only be used on linux - please check that your code does not call it directly.\n".postf(thisMethod);
	}
}
```
yes?

**Comment by brianlheim** [comment]
I'd prefer it to be a no-op with an error, or at least a message with with `.warn`. In any case, `thisProcess.platform` returns a class instance on my machine, so I think the proper test is
```
if(thisProcess.platform.name != 'linux') {
```

**Comment by brianlheim** [comment]
Ping. @adcxyz would you mind making one last commit to change `.getClientID`? It's not a big deal to me how specifically the no-op works, but it would be nice to have. After that, I think we're ready to merge.

**Comment by adcxyz** [comment]
ok, done, ready to go :-)

**Comment by adcxyz** [comment]
sorry, accidentally committed help file update to master instead of topic help fixes. 
better to fix when merging, or before?

File: SCClassLibrary/Common/Control/MIDIOut.sc [hunk 8]
```diff
@@ -94,6 +94,16 @@ MIDIClient {
 	*externalDestinations{
 		^destinations;
 	}
+
+	*getClientID {
```
Comment (telephon) [inline SCClassLibrary/Common/Control/MIDIOut.sc]: should this method maybe be named `pr_getClientID`?

**Comment by jamshark70** [comment]
Now the branch has conflicts with the new stuff in master.

Probably the best is to merge or rebase the topic branch against the new stuff in master, resolve conflicts, and `git push --force` the topic branch.

If the new stuff in master is newer than the topic branch, merge master into the topic.

If master is older, rebase.

**Comment by adcxyz** [comment]
ok, resolving the conflict worked on github editor directly - should be fine now.

File: SCClassLibrary/Common/Control/MIDIOut.sc (hunk 8 above)
Comment (adcxyz) [inline SCClassLibrary/Common/Control/MIDIOut.sc]: semantically fine with me, but would be an API change. 
don't know if it ever occurs in user code on linux - does it?

Comment (brianlheim) [inline SCClassLibrary/Common/Control/MIDIOut.sc]: could be renamed, but that's not the focus of this PR

**Comment by brianlheim** [comment]
I applied the swing/Qt change @vivid-synth mentioned above, and deleted the .split method you added, @adcxyz.

AFAICT, SequenceableCollection:-split was added by @telephon and reverted in  bf2f488a43c716854e79b56584a9bfbc3beb3b80, but its corresponding documentation was never deleted. If you want to add it, that's fine, but this PR is about fixing documentation and a new method going in SeqColl should probably have its own PR w/ unit tests.

I take similar issue with the addition of stub ImageFilter and ImageKernel classes and 11 stub methods to Image. The better solution IMO would be to eliminate references to these classes and methods from the helpfiles.

**Comment by telephon** [comment]
what's left to be done here?

File: HelpSource/Classes/ScrollView.schelp (hunk 5 above)
Comment (telephon) [inline HelpSource/Classes/ScrollView.schelp]: Yes, for sc files and documentation, we do.

**Comment by brianlheim** [comment]
@telephon Please don't merge PRs that haven't been approved or reached consensus.

**Comment by telephon** [comment]
@brianlheim why was there no consensus? The disputed `split` method was reverted, and as it seemed to me, all other requests were followed. There was no such label, either.

**Comment by brianlheim** [comment]
@telephon https://github.com/supercollider/supercollider/pull/2606#issuecomment-279196899

**Comment by telephon** [comment]
Ah ok, I didn't understand this. 

So I'll go and remove:

> stub ImageFilter and ImageKernel classes and 11 stub methods to Image.

and the `split` documentation.

**Comment by brianlheim** [comment]
@telephon Thank you!

**Comment by telephon** [comment]
>I take similar issue with the addition of stub ImageFilter and ImageKernel classes and 11 stub methods to Image. The better solution IMO would be to eliminate references to these classes and methods from the helpfiles.

Sorry, I don't understand, because I've never used them: should we remove the classes?

>and the split documentation.

was already removed.

**Comment by adcxyz** [comment]
dear all, 
yes please just remove the Image classes and methods. The stubs were not a good idea, and I doubt that keeping them around would attract any devs to reimplement these.

**Pull request merged**